apps/command-palette-wp-admin/src/index.js

```diff
@@ -0,0 +1,73 @@
+import CommandPalette, {
+	useAtomicCommands,
+	useAtomicLimitedCommands,
+	useWpcomSimpleSiteLimitedCommands,
+	useWpcomSimpleSiteCommands,
+} from '@automattic/command-palette';
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import domReady from '@wordpress/dom-ready';
+import { render } from 'react-dom';
+import WPCOM from 'wpcom';
+import proxyRequest from 'wpcom-proxy-request';
+
+function CommandPaletteApp() {
+	const wpcom = new WPCOM( proxyRequest );
+	wpcom.request( { metaAPI: { accessAllUsersBlogs: true } } );
+
+	if ( ! window.commandPaletteConfig ) {
+		// Can't load the command palette without a config.
+		return null;
+	}
+
+	const currentRoute = window.location.pathname;
+	const siteHostname = window.location.hostname;
+
+	const navigate = ( path, openInNewTab ) => {
+		if ( path.startsWith( '/' ) && ! path.startsWith( '/wp-admin' ) ) {
+			path = `https://wordpress.com${ path }/${ siteHostname }`;
+		}
+
+		window.open( path, openInNewTab ? '_blank' : '_self' );
+	};
+
+	const {
+		isAdmin = false,
+		isAtomic = false,
+		isSelfHosted = false,
+		isSimple = false,
+	} = window?.commandPaletteConfig || {};
+
+	let useCommands;
+	if ( isAtomic && ! isSelfHosted ) {
+		useCommands = isAdmin ? useAtomicCommands : useAtomicLimitedCommands;
+	}
+
+	if ( isSimple ) {
+		useCommands = isAdmin ? useWpcomSimpleSiteCommands : useWpcomSimpleSiteLimitedCommands;
+	}
+
+	if ( ! useCommands ) {
+		// Can't load the command palette without a valid commands provider function.
+		return null;
+	}
```

Comment:
Just wanted to follow up on https://github.com/Automattic/wp-calypso/pull/87790#discussion_r1500394392, but this is not a blocker for this PR.

As a reminder, we're right now quintupling the commands in several files:

- `wpcom-smp-commands.tsx`: Commands for Calypso
- `use-atomic-commands.tsx`: Commands for users with the `manage_options` cap of Atomic sites in WP Admin
- `use-limited-atomic-commands.tsx`: Commands for users without the `manage_options` cap of Atomic sites in WP Admin
- `use-wpcom-simple-site-commands.tsx`: Commands for users with the `manage_options` cap of Simple sites in WP Admin
- `use-wpcom-simple-site-limited-commands.tsx`: Commands for users without the `manage_options` cap of Simple sites in WP Admin

But this system won't work for our needs, since we also have commands for other capabilities (`edit_post`, `media_upload`, etc) and for other sites (P2, private, staging, etc), so this system won't scale.

We need a different system that leverages on the existing `siteFunctions.capabilityFilter` and `siteFunctions.filter` definitions, so we can have a single source of truth for the commands instead of maintaining duplicated lists of commands elsewhere.

Reply:
Not sure if it's directly useful but the [GB implementation](https://github.com/WordPress/gutenberg/blob/trunk/packages/commands/README.md#static-commands) kinda has two ways of doing this:

 1. The "command loaders" that return a list of commands get the command palette search term as a prop (& could query the site or user or whatever) and return commands that are valid
 2. The command loaders optionally register themselves with "contexts" and it's up to the palette to call the loaders associated with given contexts.

(There's also an undocumented third way where [if your command has a `disabled` property it won't be registered](https://github.com/WordPress/gutenberg/blob/702f78adc17295bef1402c7577c07ee4aa6ba028/packages/commands/src/hooks/use-command.js#L41))